src/vmm_core.mli

```diff
@@ -123,30 +123,56 @@ module Unikernel : sig
 
   val restart_handler : config -> bool
 
+  type arguments = {
+    fail_behaviour : fail_behaviour;
+    cpuid : int ;
+    memory : int ;
+    block_devices : (string * string option * int option) list ;
+    bridges : (string * string option * Macaddr.t option) list ;
+    argv : string list option ;
+  }
+
+  val pp_arguments : arguments Fmt.t
+
+  val pp_arguments_with_argv : arguments Fmt.t
+
   type t = {
     config : config;
     cmd : string array;
     pid : int;
-    taps : string list;
+    taps : (string * Macaddr.t) list;
     digest : string;
     started : Ptime.t;
   }
 
   val pp : t Fmt.t
 
+  type block_info = {
+    unikernel_device : string ;
+    host_device : string ;
+    sector_size : int ;
+    size : int ;
+  }
+
+  type net_info = {
+    unikernel_device : string ;
+    host_device : string ;
+    mac : Macaddr.t ;
+  }
+
   type info = {
     typ : typ ;
     fail_behaviour : fail_behaviour;
     cpuid : int ;
     memory : int ;
-    block_devices : (string * string option * int option) list ;
-    bridges : (string * string option * Macaddr.t option) list ;
+    block_devices : block_info list ;
+    bridges : net_info list ;
     argv : string list option ;
     digest : string ;
     started : Ptime.t ;
   }
```

Comment:
I solved some confusion (hopefully) by moving to records for block and network devices -- they also no longer have any optional fields, but are completely filled.